docs/docsite/rst/dev_guide/developing_modules.rst

```diff
@@ -1,110 +1,63 @@
-as we work through this tutorial.
-
-See :ref:`all_modules` for a list of existing modules.
-
-Modules can be written in any language and are found in the path specified
-by :envvar:`ANSIBLE_LIBRARY` or the ``--module-path`` command line option or
-in the :envvar:`library section of the Ansible configuration file <ANSIBLE_LIBRARY>`.
-
 .. _module_dev_should_you:
 
+****************************
 Should You Develop A Module?
-````````````````````````````
-Before diving into the work of creating a new module, you should think about whether you actually *should*
-develop a module. Ask the following questions:
+****************************
+
+Developing Ansible modules is easy, but often it isn't necessary. Before you start writing a new module, ask:
 
 1. Does a similar module already exist?
 
-There are a lot of existing modules available, you should check out the list of existing modules at :ref:`modules`
+An existing module may cover the functionality you want. Ansible Core includes thousands of modules. Search our :ref:`list of existing modules <all_modules>` to see if there's a module that does what you need.
 
-2. Has someone already worked on a similar Pull Request?
+2. Does a Pull Request already exist?
 
-It's possible that someone has already started developing a similar PR. There are a few ways to find open module Pull Requests:
+An existing Pull Request may cover the functionality you want. If someone else has already started developing a similar module, you can review and test it. There are a few ways to find open module Pull Requests:
 
 * `GitHub new module PRs <https://github.com/ansible/ansible/labels/new_module>`_
 * `All updates to modules <https://github.com/ansible/ansible/labels/module>`_
 * `New module PRs listed by directory <https://ansible.sivel.net/pr/byfile.html>`_ search for `lib/ansible/modules/`
 
-If you find an existing PR that looks like it addresses the issue you are trying to solve, please provide feedback on the PR -  this will speed up getting the PR merged.
+If you find an existing PR that looks like it addresses your needs, please provide feedback on the PR -  this will speed up the review and merge process.
 
 3. Should you use or develop an action plugin instead?
 
-Action plugins get run on the master instead of on the target. For modules like file/copy/template, some of the work needs to be done on the master before the module executes on the target. Action plugins execute first on the master and can then execute the normal module on the target if necessary.
-
-For more information about action plugins, read the :ref:`action plugins documentation <developing_plugins>`.
+An action plugin may be the best way to get the functionality you want. Action plugins run on the master instead of on the target, and their functionality is available to all modules. For more information about developing plugins, read the :ref:`developing plugins page <developing_plugins>`.
```

Comment:
Change 'master' to 'control node'. I believe that is the name we use for "the machine running Ansible to configure other things".

Reply:
Plus this avoids the rather negative terms 'master/slave'.

Reply:
I was trying to find a definition of the different plugin types and could not. We mention them [here](https://docs.ansible.com/ansible/latest/plugins/action.html) but don't really spell out what each type of plugin is and how they are executed (we do explain some of them, just not all). Maybe this is something we could add to the [Developing Plugins](https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html) section